go.mod

```diff
@@ -1,19 +1,15 @@
 module github.com/aliyun/alibaba-cloud-sdk-go
 
+go 1.12
```

Comment:
Please remove version restrictions。

Reply:
It's not actually a version restriction:

```
The go directive in a go.mod file now indicates the version of the language used by the files within that module. It will be set to the current release (go 1.12) if no existing version is present. If the go directive for a module specifies a version newer than the toolchain in use, the go command will attempt to build the packages regardless, and will note the mismatch only if that build fails. 
```

This will work equally on Go 1.10 and Go 1.11 as well. Later versions of the compiler will complain though:

For example, a Go 1.13 compiler would complain if this was set to `1.15`.

Later versions of the go tool adds the directive by default, so one way or another it will make it's way to `go.mod`.

Setting it to 1.12 is actually the safest thing to do at the moment: eariler versions of the compiler won't do anything with it, every later versions will work fine with it.

Reply:
As you say，eariler versions of the compiler won't do anything with it. If you set it to 1.12, It may not work on 1.11.

Reply:
Earlier versions of the compiler ignore that directive entirely, it won't change the behavior of the compiler.

> If you set it to 1.12, It may not work on 1.11.

If that were true, I guess travis builds should have broken, but they haven't.

Here is a list of projects with that directive (Go 1.10 and 1.11 are in the build targets):

https://github.com/aliyun/aliyun-cli/blob/master/go.mod#L3

https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/go.mod#L42

Reply:
We can't decide which go version the users use, they may use Go 1.11 and want to use go mod at the same time. Aliyun-Cli and Terraform are not dependencies for users, so they don't care which go version the users use.

Reply:
> Earlier versions of the compiler ignore that directive entirely, it won't change the behavior of the compiler.
> 
> > If you set it to 1.12, It may not work on 1.11.
> 
> If that were true, I guess travis builds should have broken, but they haven't.
> 
> Here is a list of projects with that directive (Go 1.10 and 1.11 are in the build targets):
> 
> https://github.com/aliyun/aliyun-cli/blob/master/go.mod#L3
> 
> https://github.com/terraform-providers/terraform-provider-alicloud/blob/master/go.mod#L42

Due to we support go 1.10 now, travis doesn't use go mod to  compile.